specs/Standards/Tokens/NonFungibleToken.md

```diff
@@ -0,0 +1,167 @@
+- If one or both of the final `transfer` calls fail, then nothing will happen and `escrow` should attempt reissuing the failed transaction.
+
+#### **Technical calls**
+
+1. `alice` makes an async call to `corgi::grant_access({"escrow_account_id":"escrow"})`
+2. `jerry`  makes an async call to `sausage::grant_access({"escrow_account_id":"escrow"})`
+3. `escrow` calls `sausage::transfer({"new_owner_id:"escrow", "token_id": 5})`
+    - Recommondation: attach callback `escrow::on_transfer({"owner_id":"jerry", "token_contract":"sausage", "token_id": 5})`
+4. `escrow` calls `corgi::transfer({"new_owner_id:"escrow", "token_id": 3})`
+    - Recommendation: attach callback `escrow::on_transfer({"owner_id":"alice", "token_contract":"corgi", "token_id": 3})`
+5. In one Promise:
+    1. `escrow` calls `corgi::transfer({"new_owner_id:"jerry", "token_id": 3})`
+        - attaches callback `escrow::on_transfer({"owner_id":"alice", "token_contract:"corgi", "token_id": 3})`
+    2. `escrow` calls `sausage::transfer({"new_owner_id:"escrow", "token_id": 5})`
+        - attaches callback `escrow::on_transfer({"owner_id":"jerry", "token_contract":"corgi", "token_id": 3})`
+
+
+## Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+## Template for smart contract in AssemblyScript
+
+At time of writing, this standard is established with several constraints found in AssemblyScript. The first is that interfaces are not an implemented feature of AssemblyScript, and the second is that classes are not exported in the conversion from AssemblyScript to WASM. This means that the entire contract could be implemented as a class, which might be better for code organization, but it would be deceiving in function.
+
+```TypeScript
+
+/******************/
+/* CHANGE METHODS */
+/******************/
+
+// Grant access to the given `escrow_account_id` for all tokens that account has.
+// Requirements:
+// * The caller of the function (`predecessor_id`) should have access to the tokens.
+export function grant_access(escrow_account_id: string): void;
+
+// Revoke the access to the given `account_id` for the given `token_id`.
+// Requirements:
+// * The caller of the function (`predecessor_id`) should have access to the token.
+export function revoke_access(escrow_account_id: string): void;
+
+// Transfer the given `token_id`. Account `new_account_id` becomes the new owner.
+// Requirements:
+// * The caller of the function (`predecessor_id`) should have access to the token.
+export function transfer(new_owner_id: string, token_id: u128): void;
+
+/****************/
+/* VIEW METHODS */
+/****************/
+
+// Returns `true` or `false` based on `escrow_account_id` having access to tokens owned by `owner_id`
+export function check_access(escrow_account_id: string, owner_id: string): boolean;
+
+// Get an individual owner by given `token_id`.
```

Comment:
```suggestion
// Get the owner's account ID of the given `token_id`.
```

Reply:
"_for_ the given `token_id`" sounds better to my ears